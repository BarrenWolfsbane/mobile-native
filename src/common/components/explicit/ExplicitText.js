import React, {
  Component
} from 'react';

import {
  NavigationActions
} from 'react-navigation';

import {observer} from "mobx-react/native";
import entities from 'entities';
import {
  Text,
  StyleSheet,
  View,
} from 'react-native';

import Tags from '../../../common/components/Tags';

@observer
export default class ExplicitText extends Component {

  render() {
    const entity = this.props.entity;
    const message = entities.decodeHTML(this.props.entity.text);
    return (
        <View style={{flex:1}}>
          { entity.mature ?
            <Text style={styles.mature}>{message}</Text>:
            <Tags navigation={this.props.navigation}>{message}</Tags>
          }
        </View>
    );
  }
}

const styles = StyleSheet.create({
  mature: {
    color:'transparent',
    textShadowColor: 'rgba(107, 107, 107, 0.5)',
    textShadowOffset: {width: -1, height: 1},
    textShadowRadius: 20
  }
});