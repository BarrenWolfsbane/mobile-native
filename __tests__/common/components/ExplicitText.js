import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';
import ExplicitText from '../../../src/common/components/explicit/ExplicitText';

import { activitiesServiceFaker } from '../../../__mocks__/fake/ActivitiesFaker';

import renderer from 'react-test-renderer';

fdescribe('Explicit text component', () => {

  let user, comments, entity, screen;
  beforeEach(() => {

    let mockResponse = activitiesServiceFaker().load(1);
    entity = mockResponse.activities[0];
    entity.mature = true;
    entity.mature_visibility = true;
    entity.decodeHTML = jest.fn();
    entity.toggleMatureVisibility = jest.fn();
    entity.decodeHTML.mockReturnValue('string');

    screen = shallow(
      <ExplicitText entity={entity}/>
    );

    jest.runAllTimers();
  });

  it('renders correctly', async () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });

  it('should have a TouchableOpacity', async () => {
    screen.update();

    expect(screen.find('TouchableOpacity')).toHaveLength(1);
  });


  it('change visibility', async () => {

    screen.update();

    screen.find('TouchableOpacity').simulate('press');
    screen.update();

    expect(entity.toggleMatureVisibility).toHaveBeenCalled();
  });
});
