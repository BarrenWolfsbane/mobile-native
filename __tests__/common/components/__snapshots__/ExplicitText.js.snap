// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explicit text component renders correctly 1`] = `
<Component
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Tags>
    undefined
  </Tags>
  <TouchableOpacity
    activeOpacity={0.2}
    onPress={[Function]}
    style={
      Array [
        Object {
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "marginTop": -15,
        },
      ]
    }
  >
    <Icon
      allowFontScaling={false}
      color="red"
      name="explicit"
      size={18}
      style={
        Object {
          "elevation": 1,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 0.5,
          },
          "shadowOpacity": 0.5,
          "shadowRadius": 0.5,
        }
      }
    />
  </TouchableOpacity>
</Component>
`;
